cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -227,13 +232,14 @@ func Run(c *config.CompletedConfig, stopCh <-chan struct{}) error {
 		controllerContext.ObjectOrMetadataInformerFactory.Start(stopCh)
 		close(controllerContext.InformersStarted)
 
-		select {}
+		<-ctx.Done()
```

Comment:
unrelated, but doesn't feel that this `run` closure has a mix of contexts and stopCh that is really confusing?
mainly because IIUIC this ctx.Done() is derived from the stopCh

Reply:
we need to translate between function that expect stopCh and those that expect ctx (and ctx.Done()).
I think that's fine